polyply/tests/test_gen_coords_logic.py

```diff
@@ -47,6 +47,51 @@ def test_no_positions_generated(tmp_path, monkeypatch):
         assert np.all(molecule_out.nodes[node]['position'] ==
                       molecule_in.nodes[node]['position'])
 
+@pytest.mark.parametrize('box_input, box_ref, density, warning', [
+                        # box from input coordinates
+                        (None, np.array([11.0, 11.0, 11.0]), None, None),
+                        # box from input coordinates overwrites
+                        (np.array([5.0, 5.0, 5.0]), np.array([11.0, 11.0, 11.0]), None, "warn1"),
+                        # box from input coordinates and density from CLI
+                        (None, np.array([11.0, 11.0, 11.0]), 1000, "warn2"),
+                        ])
+def test_box_input(tmp_path, monkeypatch, caplog, box_input, box_ref, density, warning):
+    """
+    Here we test that the correct box is chosen, in case there
+    are conflicting inputs.
+    """
+    warnings = {"warn1": ("A box is provided via the -box command line "
+                          "and the starting coordinates. We consider the "
+                          "the box of starting coordinates as correct. "),
+                "warn2": ("A density is provided via the command line, "
+                          "but the starting coordinates define a box."
+                          "Will try to pack all molecules in the box "
+                          "provied with starting coordinates."),}
```

Comment:
Because this tests the logic box is in both the warnings. And it's not good enough to just get a warning. It also needs to give the appropriate warning. I agree it's not super elegant but does the job

Reply:
Sure. But I think it's better to pick a specific phrase from the warning to check if that exists. Currently the warning also has a typo for example (just saw it), which now needs to be fixed in multiple places. Also if you want to reword or clarify a warning further down the line you run into a test failure without an obvious reason.

Reply:
I really fail to see the point. Sure you need to change it in two places now, but the argument you made can be reversed. If I look for a somewhat specific part in the warning and later some other warning with that same partial string gets integrated the test might incorrectly pass.